packages/block-editor/src/components/list-view/index.js

```diff
@@ -50,15 +50,22 @@ export const BLOCK_LIST_ITEM_HEIGHT = 36;
 /**
  * Show a hierarchical list of blocks.
  *
- * @param {Object}  props                 Components props.
- * @param {string}  props.id              An HTML element id for the root element of ListView.
- * @param {Array}   props.blocks          Custom subset of block client IDs to be used instead of the default hierarchy.
- * @param {boolean} props.showBlockMovers Flag to enable block movers
- * @param {boolean} props.isExpanded      Flag to determine whether nested levels are expanded by default.
- * @param {Object}  ref                   Forwarded ref
+ * @param {Object}  props                          Components props.
+ * @param {boolean} props.allowRightClickOverrides Flag to enable right click contextual menu.
+ * @param {string}  props.id                       An HTML element id for the root element of ListView.
+ * @param {Array}   props.blocks                   Custom subset of block client IDs to be used instead of the default hierarchy.
+ * @param {boolean} props.showBlockMovers          Flag to enable block movers
+ * @param {boolean} props.isExpanded               Flag to determine whether nested levels are expanded by default.
+ * @param {Object}  ref                            Forwarded ref
  */
 function ListView(
-	{ id, blocks, showBlockMovers = false, isExpanded = false },
+	{
+		allowRightClickOverrides,
```

Comment:
I personally don't think the preference is needed to trial this. It's the kind of thing that could be added to the plugin with the goal of seeking feedback.

It'd also be good to avoid introducing new stable props as I think ListView should move towards become a stable API soon. (some discussion here - https://github.com/WordPress/gutenberg/pull/40777)

An alternative if the preference is required could be making this an editor setting and pulling it in using `useSelect`.

Reply:
Thanks for taking a look @talldan! It'll be easy enough to strip out the setting.

The main reason I included it is that we don't yet have a precedent in Gutenberg (that I could find, at least 🙂) to override the browser default behaviour, and I imagined that there could be some folks that might feel strongly about whether or not it's okay to override right click. I'll be keen to hear if anyone feels strongly about it!

> An alternative if the preference is required could be making this an editor setting and pulling it in using useSelect.

If we do keep the setting, that does sound like a better way to do it, it felt a bit awkward adding it as a stable prop (and all the prop drilling involved). I might not get a chance today to switch over to that approach, but can take a look on Monday.

Thanks again!

Reply:
I've moved it over to an editor setting in https://github.com/WordPress/gutenberg/pull/41041/commits/8fd68db981b5df3717a5b22edb471d0a3f52c9c1 — that makes the change to `ListView` much neater, without the additional prop drilling 👍

Happy to remove the setting, too, if need be, but 🤞 this PR should be in a better place for gathering more feedback now. Cheers! 🙇